hub/core/tensor.py

```diff
@@ -117,13 +123,51 @@ def extend_tensor(array: np.ndarray, key: str, storage: StorageProvider, **kwarg
         raise TensorDoesNotExistError(key)
 
     tensor_meta, index_meta = _get_metas_from_kwargs(key, storage, **kwargs)
-
     # extend is guaranteed to have a batch axis
     tensor_meta.check_batch_is_compatible(array)
 
     write_array(array, key, storage, tensor_meta, index_meta)
 
 
+def add_index_map_to_tensor(
+    index_map_dict: dict, key: str, storage: StorageProvider, **kwargs
```

Comment:
another note: please remove all `index_map` references and replace them with `index_meta`